File: Java/queue/llama_3_1_8b.java
package Java.queue;
import java.util.NoSuchElementException;

public class llama_3_1_8b {
   
    private Node<T> head;
    private Node<T> tail;
    private int size;

    private static class Node<T> {
        T data;
        Node<T> next;

        public Node(T data) {
            this.data = data;
            this.next = null;
        }
    }

    public llama_3_1_8b() {
        head = null;
        tail = null;
        size = 0;
    }

    public void enqueue(T data) {
        Node<T> newNode = new Node<>(data);
        if (tail == null) {
            head = newNode;
            tail = newNode;
        } else {
            tail.next = newNode;
            tail = newNode;
        }
        size++;
    }

    public T dequeue() {
        if (head == null) {
            throw new NoSuchElementException("Queue is empty");
        }
        T data = head.data;
        head = head.next;
        if (head == null) {
            tail = null;
        }
        size--;
        return data;
    }

    public T peek() {
        if (head == null) {
            throw new NoSuchElementException("Queue is empty");
        }
        return head.data;
    }

    public int size() {
        return size;
    }

    public boolean isEmpty() {
        return head == null;
    }
}

File: Python/queue/llama-3-1-8b.py
from collections import deque

class Queue:
    """
    A class representing a queue data structure.
    
    Attributes:
    ----------
    queue : deque
        A double-ended queue to store elements.
        
    Methods:
    -------
    enqueue(element)
        Adds an element to the end of the queue.
    dequeue()
        Removes an element from the front of the queue.
    peek()
        Returns the element at the front of the queue without removing it.
    is_empty()
        Checks if the queue is empty.
    size()
        Returns the number of elements in the queue.
    """

    def __init__(self):
        """
        Initializes an empty queue.
        """
        self.queue = deque()

    def enqueue(self, element):
        """
        Adds an element to the end of the queue.
        
        Parameters:
        ----------
        element : any
            The element to be added to the queue.
        """
        self.queue.append(element)

    def dequeue(self):
        """
        Removes an element from the front of the queue.
        
        Returns:
        -------
        any
            The removed element.
        
        Raises:
        ------
        IndexError
            If the queue is empty.
        """
        if self.is_empty():
            raise IndexError("Cannot dequeue from an empty queue")
        return self.queue.popleft()

    def peek(self):
        """
        Returns the element at the front of the queue without removing it.
        
        Returns:
        -------
        any
            The element at the front of the queue.
        
        Raises:
        ------
        IndexError
            If the queue is empty.
        """
        if self.is_empty():
            raise IndexError("Cannot peek an empty queue")
        return self.queue[0]

    def is_empty(self):
        """
        Checks if the queue is empty.
        
        Returns:
        -------
        bool
            True if the queue is empty, False otherwise.
        """
        return len(self.queue) == 0

    def size(self):
        """
        Returns the number of elements in the queue.
        
        Returns:
        -------
        int
            The number of elements in the queue.
        """
        return len(self.queue)
